app/models/activity.rb

```diff
@@ -14,6 +14,7 @@ class Activity < ApplicationRecord
   belongs_to :activity, optional: true
   has_many :activities, foreign_key: "activity_id"
   belongs_to :organisation
+  belongs_to :extending_organisation, foreign_key: "extending_organisation_id", class_name: "Organisation", optional: true
```

Comment:
It should have been optional from the time it was introduced. If it was required I would expect the wizard steps to fail validation as no extending org has been set on the Activity?

Reply:
I was meant to organise my note here into something more meaningful but missed it clearly! 

I think I saw this screenshot which looked like it is a mandatory field https://user-images.githubusercontent.com/480578/74239049-fc69fd80-4cce-11ea-90f8-16b4f74f2546.png.

Does this code start returning a validation error if the `extending_organisation_id` is blank? 

```
    @activity.assign_attributes(extending_organisation_id: extending_organisation_id)
    if @activity.valid?(:update_extending_organisation)
```

Reply:
Yep, that is the intention, if `extending_oganisation_id` is blank, the Activity is invalid through the `update_extending_organisation` validation context:

https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/de6e676b3980aa23b1c0bcc8d3a138ac1152bd24/app/models/activity.rb#L13

It's an odd one this. I'll give explaining the thinking a go, see how it comes accross:

- We have an Activity, otherwise we cannot add the extending organisaiton.
- We set the extending organisation with a radio button in the UI
- when the user first 'edits' or 'sets' the `extending_organisation` no radio is selected because no extending_oganisation associatons exists.
- At this point the `extending_organisation` is required through the validation context `update_extending_organisation` so if the user tries to submit the edit form without a selected extending_organisation they see the error

I guess the alternative is to let the user leave the `extending_organisation` unset? Which is fine if we think that should be the behaviour?

Reply:
Ah thank you for explaining, this makes sense now I see how it works in tandem with that custom validation!